File: 01_Language/01_Functions/python/mktime.py
# coding: utf-8

import datetime


def mktime(hour, minute, second, month, day, year):
    return datetime.datetime(year, month, day, hour, minute, second).timestamp()


if __name__ == '__main__':
    import os

    os.environ['TZ'] = 'UTC'
    print(mktime(0, 0, 0, 7, 1, 2000))

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Mktime.java
package com.onevgo.functions;

import java.time.LocalDateTime;
import java.time.ZoneId;
import java.util.Date;

public class Mktime {
    public static long mktime(int hour, int minute, int second, int month, int day, int year) {
        return LocalDateTime.of(year, month, day, hour, minute, second).atZone(ZoneId.systemDefault()).toEpochSecond();
    }

    public static void main(String[] args) {
        System.out.println(new Date(mktime(0, 0, 0, 1, 1, 2020) * 1000));
    }
}
